### ISSUE TITLE
Order of media queries in v5 (breakpoints object, defaultProps)

### ISSUE THREAD

**Issue by fk** [styled-system/styled-system#551]
Using the latest v5, I ran into an issue of media queries being ordered wrong when used with a `breakpoints` _object_ and `defaultProps`. This worked fine in v4.

Here's a https://codesandbox.io/s/styled-system-defaultprops-breakpoints-410v5 with a reproduction. I'll hold back from explaining what is going on there for now (in the hope that the sandbox code is self explanatory).

<img width="1341" alt="foo" src="https://user-images.githubusercontent.com/21834/59212900-da563380-8bb3-11e9-9bf1-509df720c524.png">

![image](https://user-images.githubusercontent.com/21834/59213062-4769c900-8bb4-11e9-9ebd-b73452df91eb.png)

**Comment by leekad** [comment]
I have the same problems, in my case:
- defaultProps now override all the props

example from `rebass`

```
const Header = styled(Text)({})

Header.defaultProps = {
  m: 0,
  mb: '8px'
}
```

So the `defaultProps` now override all `m`, `mb` when I create <Header mb={10} />

**Comment by sarneeh** [comment]
I have the same issue but actually in `v4`. Unfortunately, for some reason, I can't reproduce it. It behaves like that in my project but not in any other environment :confused:

**Comment by jxnblk** [comment]
I *think* this was resolved in `v5.0.18`, but feel free to reopen this if you still notice the issue:
https://codesandbox.io/s/styled-system-defaultprops-breakpoints-bs2gf
